custom_components/spook/repairs.py

```diff
@@ -188,10 +192,9 @@ async def _async_call_inspect_debouncer(_: Event) -> None:
         if self.inspect_on_reload:
 
             @callback
-            def _filter_event(event_data: Mapping[str, Any] | Event) -> bool:
+            def _filter_event(data: Mapping[str, Any] | Event) -> bool:
                 """Filter for reload events."""
-                if type(event_data) is Event:  # pylint: disable=unidiomatic-typecheck
-                    event_data = event_data.data
+                event_data = data.data if isinstance(data, Event) else data
```

Comment:
isinstance is pretty expensive in the ven handling, and avoided for that reason

Reply:
Can change that, just not sure it's worth it.
```py
from timeit import timeit

setup = """\
class Event: ...
data = Event()
"""

print(f"isinstance: {timeit('isinstance(data, Event)', setup=setup, number=10**8)}")
print(f"type: {timeit('type(data) is Event', setup=setup, number=10**8)}")
```
```
isinstance: 1.0133735409472138
type: 1.0018806660082191
```
Tested with Python 3.12

Reply:
Makes sense 👍